northstar/src/runtime/minijail.rs

```diff
@@ -69,16 +70,25 @@ impl<'a> Minijail<'a> {
         })?;
         let mut lines = io::BufReader::new(async_reader).lines();
 
+        let stop_token = CancellationToken::new();
+        let task_stop = stop_token.clone();
+
         // Spawn a task that forwards logs from minijail to the rust logger.
         let log_task = task::spawn(async move {
-            while let Ok(Some(line)) = lines.next_line().await {
-                let l = line.split_whitespace().skip(2).collect::<String>();
-                match line.chars().next() {
-                    Some('D') => debug!("{}", l),
-                    Some('I') => info!("{}", l),
-                    Some('W') => warn!("{}", l),
-                    Some('E') => error!("{}", l),
-                    _ => trace!("{}", line),
+            loop {
+                select! {
+                    Ok(Some(line)) = lines.next_line() => {
+                        let l = line.split_whitespace().skip(2).collect::<String>();
+                        match line.chars().next() {
+                            Some('D') => debug!("{}", l),
+                            Some('I') => info!("{}", l),
+                            Some('W') => warn!("{}", l),
+                            Some('E') => error!("{}", l),
+                            _ => trace!("{}", line),
+                        }
+                    }
+                    _ = task_stop.cancelled() => break,
+                    else => break,
```

Comment:
A nice pattern is to avoid the need to find a name doesn't fit (just because a clone is needed). Put it into braces. Clone everything needed on the top. This also underlines what elements are moved into the task.
```rust
{
   let stop_token = stop_token.clone();
   ....
}
```

Reply:
Sure, but this function has already too many levels :sweat_smile:

Reply:
But short ones ;-)

Reply:
as you wish :)